src/config.rs

```diff
@@ -1,160 +1,520 @@
-    MailServer,
-    MailUser,
-    MailPw,
-    MailPort,
-    ImapCertificateChecks,
-    SendServer,
-    SendUser,
-    SendPw,
-    SendPort,
-    SmtpCertificateChecks,
-    ServerFlags,
-    #[strum(props(default = "INBOX"))]
-    ImapFolder,
-    Displayname,
-    Selfstatus,
-    Selfavatar,
-    #[strum(props(default = "1"))]
-    BccSelf,
-    #[strum(props(default = "1"))]
-    E2eeEnabled,
-    #[strum(props(default = "1"))]
-    MdnsEnabled,
-    #[strum(props(default = "1"))]
-    InboxWatch,
-    #[strum(props(default = "1"))]
-    SentboxWatch,
-    #[strum(props(default = "1"))]
-    MvboxWatch,
-    #[strum(props(default = "1"))]
-    MvboxMove,
-    #[strum(props(default = "0"))] // also change ShowEmails.default() on changes
-    ShowEmails,
-    SaveMimeHeaders,
-    ConfiguredAddr,
-    ConfiguredMailServer,
-    ConfiguredMailUser,
-    ConfiguredMailPw,
-    ConfiguredMailPort,
-    ConfiguredMailSecurity,
-    ConfiguredImapCertificateChecks,
-    ConfiguredSendServer,
-    ConfiguredSendUser,
-    ConfiguredSendPw,
-    ConfiguredSendPort,
-    ConfiguredSmtpCertificateChecks,
-    ConfiguredServerFlags,
-    ConfiguredSendSecurity,
-    ConfiguredE2EEEnabled,
-    Configured,
-    // Deprecated
-    #[strum(serialize = "sys.version")]
-    SysVersion,
-    #[strum(serialize = "sys.msgsize_max_recommended")]
-    SysMsgsizeMaxRecommended,
+#[strum_discriminants(
+    name(ConfigKey),
+    derive(Display, EnumString),
+    strum(serialize_all = "snake_case")
+)]
+pub enum ConfigItem {
+    Addr(String),
+    BccSelf(String),
+    Configured(String),
+    ConfiguredAddr(String),
+    ConfiguredE2EEEnabled(String),
+    ConfiguredImapCertificateChecks(String),
+    ConfiguredMailPort(String),
+    ConfiguredMailPw(String),
+    ConfiguredMailSecurity(String),
+    ConfiguredMailServer(String),
+    ConfiguredMailUser(String),
+    ConfiguredSendPort(String),
+    ConfiguredSendPw(String),
+    ConfiguredSendSecurity(String),
+    ConfiguredSendServer(String),
+    ConfiguredSendUser(String),
+    ConfiguredServerFlags(String),
+    ConfiguredSmtpCertificateChecks(String),
+    Displayname(String),
+    E2eeEnabled(String),
+    ImapCertificateChecks(String),
+    ImapFolder(String),
+    InboxWatch(String),
+    MailPort(String),
+    MailPw(String),
+    MailServer(String),
+    MailUser(String),
+    MdnsEnabled(String),
+    MvboxMove(String),
+    MvboxWatch(String),
+    SaveMimeHeaders(String),
+    Selfavatar(String),
+    Selfstatus(String),
+    SendPort(String),
+    SendPw(String),
+    SendServer(String),
+    SendUser(String),
+    SentboxWatch(String),
+    ServerFlags(String),
+    ShowEmails(String),
+    SmtpCertificateChecks(String),
     #[strum(serialize = "sys.config_keys")]
-    SysConfigKeys,
+    SysConfigKeys(String),
+    #[strum(serialize = "sys.msgsize_max_recommended")]
+    SysMsgsizeMaxRecommended(String),
+    #[strum(serialize = "sys.version")]
+    SysVersion(String), // Deprecated
```

Comment:
Is only `sys.version` deprecated, or are all Sys keys deprecated?

Reply:
I assumed only `SysVersion`, I just copied that bit from the existing code.  Perhaps it makes sense to also clean up the other `Sys*` keys, it would simplify things a bit (though push it to the FFI code).

I guess `SysConfigKeys` is already available via enumerating the `ConfigKey` enum.  `DC_VERSION_STR` would replace `SysVersion` and I guess only `SysMsgsizeMaxRecommended` would need to find a new home?  Not really sure how that's meant to be used.

Reply:
In existing code the "// Deprecated" comment was on its own separate line before all the Sys

Reply:
not sure where, this `// Deprecated` comes from, however, i think _none_ of the `sys.` config values are deprecated from the view of the dc_get_config() ffi function. they are used and needed.

however, in core-c the dc_get_version_str() function was deprecated - in favor to dc_get_conig("sys.version"). maybe the confusion comes from there :)

Reply:
ah, cool.  i'll remove the comment then :)  comments often moved around a bit in the c-to-rust translation